packages/client/components/GitLabScopingSearchBar.tsx

```diff
@@ -1,63 +1,89 @@
-  fontSize: ICON_SIZE.MD24,
-  padding: 3,
-  marginRight: 12
-})
-
-const SearchBar = styled('div')({
-  padding: 16
-})
-
-const SearchBarWrapper = styled('div')({
-  alignItems: 'center',
-  border: `1px solid ${PALETTE.SLATE_400}`,
-  borderRadius: '40px',
-  display: 'flex',
-  height: 44,
-  padding: '0 16px',
-  width: '100%'
-})
+import ScopingSearchBar from './ScopingSearchBar'
+import ScopingSearchHistoryToggle from './ScopingSearchHistoryToggle'
+import ScopingSearchInput from './ScopingSearchInput'
 
 interface Props {
   meetingRef: GitLabScopingSearchBar_meeting$key
 }
 
 const GitLabScopingSearchBar = (props: Props) => {
   const {meetingRef} = props
+  const atmosphere = useAtmosphere()
+  const {viewerId} = atmosphere
 
   const meeting = useFragment(
     graphql`
       fragment GitLabScopingSearchBar_meeting on PokerMeeting {
-        ...GitLabScopingSearchInput_meeting
         ...GitLabScopingSearchFilterToggle_meeting
-        ...GitLabScopingSearchCurrentFilters_meeting
+        id
+        gitlabSearchQuery {
+          selectedProjectsIds
+          queryString
+        }
+        viewerMeetingMember {
+          teamMember {
+            integrations {
+              gitlab {
+                projects {
+                  ... on _xGitLabProject {
+                    id
+                    fullPath
+                  }
+                }
+              }
+            }
+          }
+        }
       }
     `,
     meetingRef
   )
 
+  const {gitlabSearchQuery, viewerMeetingMember, id: meetingId} = meeting
+  const {selectedProjectsIds, queryString} = gitlabSearchQuery
+  const projects = viewerMeetingMember?.teamMember.integrations.gitlab.projects
+
+  const selectedProjectsPaths = [] as string[]
+  selectedProjectsIds?.forEach((projectId) => {
+    const selectedProjectPath = projects?.find((project) => project.id === projectId)?.fullPath
+    if (selectedProjectPath) selectedProjectsPaths.push(selectedProjectPath)
+  })
+  const currentFilters = selectedProjectsPaths.length ? selectedProjectsPaths.join(', ') : 'None'
+
+  const trackEvent = (eventTitle: string) => {
+    SendClientSegmentEventMutation(atmosphere, eventTitle, {
+      viewerId,
+      meetingId,
+      service: 'gitlab'
+    })
+  }
+
   return (
-    <SearchBar>
-      <SearchBarWrapper>
-        <SearchIcon>search</SearchIcon>
-        <GitLabScopingSearchInput meetingRef={meeting} />
-        <GitLabScopingSearchFilterToggle meetingRef={meeting} />
-      </SearchBarWrapper>
-      <GitLabScopingSearchCurrentFilters meetingRef={meeting} />
-    </SearchBar>
+    <ScopingSearchBar currentFilters={currentFilters}>
+      <ScopingSearchHistoryToggle />
+      <ScopingSearchInput
+        placeholder={'Search GitLab issues...'}
+        queryString={queryString}
+        meetingId={meetingId}
+        linkedRecordName={'gitlabSearchQuery'}
+        onChange={() => {
+          if (!queryString) {
+            trackEvent('Started Poker Scope Search')
```

Comment:
+1 could we always track this event? Then we could get rid of these callbacks.

Reply:
@Dschoordsch this was only added for GitLab based on this discussion here: #6296
```
SendClientSegmentEventMutation(atmosphere, eventTitle, {
      viewerId,
      meetingId,
      service: 'gitlab'
    })
   ```
   
I can track this for all integrations. But I'm not sure if we really need these events for other integrations. Should we?
 
cc @nickoferrall

Reply:
@Dschoordsch fixed